sdk/core/azure-core/src/main/java/com/azure/core/credential/TokenRefreshOptions.java

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.credential;
+
+import java.time.Duration;
+
+/**
+ * The options to configure the token refresh behavior.
+ */
+public class TokenRefreshOptions {
+    private static final Duration DEFAULT_REFRESH_RETRY_TIMEOUT = Duration.ofSeconds(30);
+    private static final Duration DEFAULT_REFRESH_OFFSET = Duration.ofMinutes(2);
+
+    private Duration tokenRefreshRetryTimeout = DEFAULT_REFRESH_RETRY_TIMEOUT;
+    private Duration tokenRefreshOffset = DEFAULT_REFRESH_OFFSET;
+
+    /**
+     * Returns a Duration value representing the amount of time to wait before retrying a token refresh. This is to
+     * prevent sending too many requests to the authentication service.
+     *
+     * @return the duration value representing the amount of time to wait before retrying a token refresh
+     */
+    public Duration getTokenRefreshRetryTimeout() {
+        return tokenRefreshRetryTimeout;
+    }
+
+    /**
+     * Specifies a Duration value representing the amount of time to wait before retrying a token refresh. This is to
+     * prevent sending too many requests to the authentication service.
+     *
+     * @param tokenRefreshRetryTimeout the amount of time to wait before retrying a token refresh
+     * @return the updated TokenRefreshOptions object
+     */
+    public TokenRefreshOptions setTokenRefreshRetryTimeout(Duration tokenRefreshRetryTimeout) {
+        this.tokenRefreshRetryTimeout = tokenRefreshRetryTimeout;
+        return this;
+    }
+
+    /**
+     * Returns a Duration value representing the amount of time to subtract from the token expiry time, whereupon
+     * attempts will be made to refresh the token. By default this will occur two minutes prior to the expiry of the
+     * token.
+     *
+     * @return the duration value representing the amount of time to subtract from the token expiry time
+     */
+    public Duration getTokenRefreshOffset() {
+        return tokenRefreshOffset;
+    }
+
+    /**
+     * Specifies the Duration value representing the amount of time to subtract from the token expiry time, whereupon
+     * attempts will be made to refresh the token. By default this will occur two minutes prior to the expiry of the
+     * token.
+     *
+     * @param tokenRefreshOffset the duration representing the amount of time to subtract from the token expiry time
+     * @return the updated TokenRefreshOptions object
+     */
+    public TokenRefreshOptions setTokenRefreshOffset(Duration tokenRefreshOffset) {
```

Comment:
same as above, could this be just `setOffset`?

Reply:
Simplified as suggested